src/lean_spec/subspecs/forkchoice/store.py

```diff
@@ -106,90 +118,118 @@ class method acts as a factory for creating a new Store instance.
         """
-        Process new attestation (signed vote).
+        Process new attestation (signed validator attestation).
 
         Handles attestations from blocks or network gossip, updating vote tracking
         according to timing and precedence rules.
 
         Args:
-            signed_vote: Attestation to process.
+            signed_attestation: Attestation to process.
             is_from_block: True if attestation came from block, False if from network.
         """
         # Validate attestation structure and constraints
-        self.validate_attestation(signed_vote)
+        self.validate_attestation(signed_attestation)
 
-        validator_id = ValidatorIndex(signed_vote.data.validator_id)
-        vote = signed_vote.data
+        attestation = signed_attestation.message
+        validator_id = ValidatorIndex(attestation.validator_id)
+        attestation_slot = attestation.data.slot
 
         if is_from_block:
             # On-chain attestation processing
 
             # Update known votes if this is the latest from validator
             latest_known = self.latest_known_votes.get(validator_id)
-            if latest_known is None or latest_known.slot < vote.slot:
-                self.latest_known_votes[validator_id] = vote.target
+            if latest_known is None or latest_known.message.data.slot < attestation_slot:
+                self.latest_known_votes[validator_id] = signed_attestation
 
             # Remove from new votes if this supersedes it
             latest_new = self.latest_new_votes.get(validator_id)
-            if latest_new is not None and latest_new.slot <= vote.target.slot:
+            if latest_new is not None and latest_new.message.data.slot <= attestation_slot:
                 del self.latest_new_votes[validator_id]
+            return
 
-        else:
-            # Network gossip attestation processing
-
-            # Ensure forkchoice is current before processing gossip
-            time_slots = self.time // SECONDS_PER_INTERVAL
-            assert vote.slot <= time_slots, "Attestation from future slot"
-
-            # Update new votes if this is latest from validator
-            latest_new = self.latest_new_votes.get(validator_id)
-            if latest_new is None or latest_new.slot < vote.target.slot:
-                self.latest_new_votes[validator_id] = vote.target
+        # Network gossip attestation processing
+
+        # Ensure forkchoice is current before processing gossip
+        time_slots = Slot(self.time // SECONDS_PER_INTERVAL)
+        assert attestation_slot <= time_slots, "Attestation from future slot"
+
+        # Update new votes if this is latest from validator
+        latest_new = self.latest_new_votes.get(validator_id)
+        if latest_new is None or latest_new.message.data.slot < attestation_slot:
+            self.latest_new_votes[validator_id] = signed_attestation
+
+    @staticmethod
+    def _is_placeholder_signature(signature: Bytes4000) -> bool:
```

Comment:
```suggestion
    def _is_valid_signature(signature: Bytes4000) -> bool:
```

Nit and subjective: imo it helps with spec readability to go with the intended name. The `TODO` comment is already self-explanatory that the implementation is not there yet.